### PR TITLE
[Remove] types from PutMappingRequest

### PR DESCRIPTION
**Pull request by nknize**
Remove type support from putMappingRequest, dependencies, and all tests.

relates #1940

### CODE REVIEW

**Comment by dreamer-89** [comment]
The latest job fails testThatUpdatingMappingShouldNotRemoveSizeMappingConfiguration test due to assertion error.
```
REPRODUCE WITH: ./gradlew ':plugins:mapper-size:internalClusterTest' --tests "org.opensearch.index.mapper.size.SizeMappingIT.testThatUpdatingMappingShouldNotRemoveSizeMappingConfiguration" -Dtests.seed=4F62945A913A057D -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=pt -Dtests.timezone=Australia/Currie -Druntime.java=17

org.opensearch.index.mapper.size.SizeMappingIT > testThatUpdatingMappingShouldNotRemoveSizeMappingConfiguration FAILED
    java.lang.AssertionError: Expected size field mapping to be enabled for foo
    Expected: map containing ["_size"->ANYTHING]
         but: map was [<properties={otherField={type=text}}>]
        at __randomizedtesting.SeedInfo.seed([4F62945A913A057D:D988F27324556493]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.opensearch.index.mapper.size.SizeMappingIT.assertSizeMappingEnabled(SizeMappingIT.java:119)
        at org.opensearch.index.mapper.size.SizeMappingIT.testThatUpdatingMappingShouldNotRemoveSizeMappingConfiguration(SizeMappingIT.java:85)
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.gradle.api.internal.tasks.testing.worker.TestWorker (file:/home/ubuntu/.gradle/wrapper/dists/gradle-7.3.3-all/4295vidhdd9hd3gbjyw1xqxpo/gradle-7.3.3/lib/plugins/gradle-testing-base-7.3.3.jar)
WARNING: Please consider reporting this to the maintainers of org.gradle.api.internal.tasks.testing.worker.TestWorker
WARNING: System::setSecurityManager will be removed in a future release
```

**Comment by dreamer-89** [comment]
start gradle check

**Comment by nknize** [comment]
There are still a few other failures as well.. Getting ready to push an update

**Comment by dreamer-89** [comment]
`TranslogPolicyIT.testRecoverReplica` failure. 
```
REPRODUCE WITH: ./gradlew ':qa:translog-policy:v1.3.0#Step1OldClusterTest' --tests "org.opensearch.upgrades.TranslogPolicyIT.testRecoverReplica" -Dtests.seed=A7F39AB3B5FF3CF9 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=mt -Dtests.timezone=Chile/Continental -Druntime.java=17

org.opensearch.upgrades.TranslogPolicyIT > testRecoverReplica FAILED
    org.opensearch.client.WarningFailureException: method [PUT], host [http://127.0.0.1:33619], URI [/testrecoverreplica], status line [HTTP/1.1 200 OK]
    {"acknowledged":true,"shards_acknowledged":true,"index":"testrecoverreplica"}
        at __randomizedtesting.SeedInfo.seed([A7F39AB3B5FF3CF9:5D6A55174026FDF4]:0)
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:346)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:320)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:295)
        at app//org.opensearch.test.rest.OpenSearchRestTestCase.createIndex(OpenSearchRestTestCase.java:976)
        at app//org.opensearch.test.rest.OpenSearchRestTestCase.createIndex(OpenSearchRestTestCase.java:956)
        at app//org.opensearch.test.rest.OpenSearchRestTestCase.createIndex(OpenSearchRestTestCase.java:952)
        at app//org.opensearch.upgrades.TranslogPolicyIT.testRecoverReplica(TranslogPolicyIT.java:129)
```

This test has started failing more frequently on Jenkins but this test is not reproducible on local machine.

**Comment by nknize** [comment]
> This test has started failing more frequently on Jenkins but this test is not reproducible on local machine

I thought I was going crazy! This is tripping up in [RestClient.java#L345](https://github.com/opensearch-project/OpenSearch/blob/ae14259a2c4b9dcf203dc46ef217563a33ae2901/client/rest/src/main/java/org/opensearch/client/RestClient.java#L345) which could be related to types removal warning but the exception doesn't print the offending warning. We should add that and try to reproduce in jenkins.

**Comment by dreamer-89** [comment]
> > This test has started failing more frequently on Jenkins but this test is not reproducible on local machine
> 
> I thought I was going crazy! This is tripping up in [RestClient.java#L345](https://github.com/opensearch-project/OpenSearch/blob/ae14259a2c4b9dcf203dc46ef217563a33ae2901/client/rest/src/main/java/org/opensearch/client/RestClient.java#L345) which could be related to types removal warning but the exception doesn't print the offending warning. We should add that and try to reproduce in jenkins.

Raised PR to log warnings https://github.com/opensearch-project/OpenSearch/pull/2338

**Comment by VachaShah** [comment]
@nknize @dreamer-89 This is related to #2334. I am working on a fix for this.

**Comment by dblock** [comment]
start gradle check

**Comment by reta** [comment]
@nknize LGTM but it looks incomplete (to me), the `PutMappingRequest` still uses `type` in payload: 

```
{
  "_doc": {
    "properties": {
      "name": {
        "type": "text",
   ....
}
```

I think it would be great to remove it as well in scope of this change, wdyt?

**Comment by nknize** [comment]
```TranslogPolicyIT``` is no longer needed. This was added to trim unsafe commits for indexes created before 6.2. This is failing now because `Step1OldClusterTest` applies to 7.x/1.x indices whereas it used to be Legacy 6.2. There's a PR open #2225 that removes all of this ancient logic.

**Comment by nknize** [comment]
Until the PR is merged I'm going to unilaterly push a MuteTest commit.

**Comment by nknize** [comment]
> I'm going to unilaterly push a MuteTest commit.

As a good steward I opened a PR instead. #2342

**Comment by reta** [comment]
> > I'm going to unilaterly push a MuteTest commit.
> 
> As a good steward I opened a PR instead. #2342

@nknize could we get https://github.com/opensearch-project/OpenSearch/pull/2225 in before merging this one?

**Comment by nknize** [comment]
> @nknize could we get #2225 in before merging this one?

Yes. It needs to land before this can be merged. 

> I think it would be great to remove it as well in scope of this change, wdyt?

I noticed a few more removals needed for this PR that I think will bring this one to ready. Then we can continue to detangle in follow up PRs.

**Review by dreamer-89 — APPROVED** [review]
LGTM!

File: client/rest-high-level/src/main/java/org/opensearch/client/IndicesClient.java [hunk 1]
```diff
@@ -456,59 +456,6 @@ public Cancellable putMappingAsync(
...
-            putMappingRequest,
-            IndicesRequestConverters::putMapping,
-            options,
-            AcknowledgedResponse::fromXContent,
-            listener,
-            emptySet()
-        );
-    }
-
     /**
```
[the remark this answers is not in the document]
Comment (nknize) [reply client/rest-high-level/src/main/java/org/opensearch/client/IndicesClient.java:459]: Yes! Upcoming PR

**Pull request merged**